src/main/java/org/generationcp/middleware/dao/UserDefinedFieldDAO.java

```diff
@@ -173,6 +173,35 @@ public List<org.generationcp.middleware.api.nametype.GermplasmNameTypeDTO> searc
 		}
 	}
 
+	public List<org.generationcp.middleware.api.nametype.GermplasmNameTypeDTO> getNameTypeFromDataset(final Integer datasetId) {
```

Comment:
1) rename to getDatasetNameTypes
2) the fully qualified name is not needed for GermplasmNameTypeDTO
3) Move to DatasetService as you are getting info related to datasets